addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -632,7 +632,13 @@ const SelectUserValueWidget = UserValueWidget.extend({
 
         this.menuTogglerEl = document.createElement('we-toggler');
         this.containerEl.appendChild(this.menuTogglerEl);
-
+        // set focus first on the 'containerEl' element for blur event
+        // because 'containerEl' elements do not receive focus by default
+        $(this.containerEl).attr('tabindex', -1);
+        // The dropdown closeup when click outside it
+        this.containerEl.onblur = (ev) => {
+            this.menuTogglerEl.classList.remove('active');
+        };
```

Comment:
There is a `closeWidgets` function that should be called for that purpose. It is already called when opening another widget or when selecting another snippet. I think the only thing you need to do is call `closeWidgets` on click inside the snippet menu (snippets.editor.js).